core_info.c

```diff
@@ -873,6 +873,33 @@ void core_info_list_get_supported_cores(core_info_list_t *core_info_list,
    *num_infos = supported;
 }
 
+/*
+ * Matches core path A and B "base" filename (ignoring everything after _libretro)
+ *
+ * Ex:
+ *   snes9x_libretro.dll and snes9x_libretro_android.so are matched
+ *   snes9x__2005_libretro.dll and snes9x_libretro_android.so are NOT matched
+ */
+bool core_info_match_base_filename(const char* core_path_a, const char* core_path_b)
+{
+   if (!core_path_a || !core_path_b)
+      return false;
+
+   {
+      const char* core_path_basename_a = path_basename(core_path_a);
+      const char* core_path_basename_b = path_basename(core_path_b);
+      const char* libretro_pos = core_path_basename_a != NULL ? strstr(core_path_basename_b, "_libretro") : NULL;
```

Comment:
`strstr()` is pretty slow - I think we can do better here by using `strrchr()` to search for the last `.`, then call it again from that position to search for the last `_`.

Also, I'd probably rename this function to `core_info_core_file_id_is_equal()`, since that's what we're checking here ('base filename' could be misconstrued as 'basename'). Maybe rename the `libretro_pos` and `core_base_filename_length` variables too :)

Reply:
Done! 
To be honest, it seems faster to just look for the latest `_`. A core filename will always contain a `.` and a `_`, so I think it's just faster to make a single `strrchr()` call..

Reply:
Ah... The issue is that a file extension may possibly contain an underscore - so a single check could yield a false positive :)

Reply:
Fixed!

Reply:
Ok fixed!